src/components/pages/coach/Anamnesis/index.jsx

```diff
@@ -102,47 +115,53 @@ class Anamnesis extends React.Component{
     componentDidMount = () => {
         // Set page title
         document.title = "Anamnese";
-        
-        // Set user variable
-        const user = this.props.location.state.user;
-        const userdata = this.props.location.state.userdata;
-        if(!this.state.user){
-            this.setState({
-                user,
-                userdata,
-                uid: user.id
-            }, () => this.initialize());
+        console.log(this.props);
+        if(this.props.match){
+            if(this.props.match.params){
+                if(this.props.match.params.uid){
+                    this.setState({
+                        uid: this.props.match.params.uid
+                    }, () => this.initialize());
+                }
+            }
         }
     }
 
     initialize = async () => {
         // Get the types and names of the anamnesis fields
         this.getAnamneseFields();
-        // Get the actual user data (if existant)
-        this.getAnmaneseData();
-        // Initialize data fetching
-        this.__init();
     }
 
     //> Fetching customer data
     __init = () => {
+        this.getBasicUserData();
+    }
+    __initExternal = (user) => {
         // Here I need to get the user city and convert it to lag/long
         // We use: https://opencagedata.com/
-        let userdata = this.state.userdata;
-        if(userdata.city && userdata.country){
-            this.__getCityLatLng(userdata.city, userdata.country);
+        if(user.city && user.country){
+            this.__getCityLatLng(user.city, user.country);
         }
     }
     __getCityLatLng = async (city, country) => {
+        console.log(city, country);
         let res = await getLatLongByCityname(city, country);
-        if(res.status === 200){
-            this.__getWeatherData(res.lat, res.lng);
-        }
+        if(res){
+            console.log("Result", res);
+            if(res.status === 200){
+                if(res.lat && res.lng){
+                    console.log("Get weather data");
+                    this.__getWeatherData(res.lat, res.lng);
+                }
+            }
+        }        
     }
     __getWeatherData = async (lat, lng) => {
+        console.log(lat, lng);
```

Comment:
# please remove `console` (no-console)

In JavaScript that is designed to be executed in the browser, it's considered a best practice to avoid using methods on `console`. Such messages are considered to be for debugging purposes and therefore not suitable to ship to the client. In general, calls using `console` should be stripped before being pushed to production.

```js
console.log("Made it here.");
console.error("That shouldn't have happened.");
```

Examples of **incorrect** code for this rule:

```js
/* eslint no-console: "error" */

console.log("Log a debug level message.");
console.warn("Log a warn level message.");
console.error("Log an error level message.");
```

Reply:
Noted.
However, the console functions do not exist without reason. It is a really great, fully integrated development tool.